### PR TITLE
feat: QC metrics for delivery

### PR DESCRIPTION
**Pull request by ivadym**
### This PR:

Adds specific metrics to `<case>_metrics_deliverables.yaml`, following the criteria described in #798 and [Atlas docs](https://github.com/Clinical-Genomics/atlas/blob/6170edaae66fa70bf25b478f2ea6ae2f0cd78572/docs/infrastructure/dataflow/workflow/metrics_delivery.md).

The workflow is activated by adding a  `--metric-delivery` flag to the `balsamic report deliver` command.

---
**Workflow**

1. `balsamic report deliver -s <case>.json --metric-delivery`
2. The metrics specified in `BALSAMIC/constants/quality_check_reporting.py ` (`METRICS_TO_DELIVER` field) are retrieved from the multiqc_data.json analysis file.
3. The output is parsed by a pydantic model and the results saved in a `<case>_metrics_deliverables.yaml`.
4. The housekeeper file to report is updated including the new deliverable file.

---
### Review and tests: 
- [ ] Tests pass
- [ ] Code review
- [ ] New code is executed and covered by tests, and test approve

### CODE REVIEW

File: BALSAMIC/constants/quality_check_reporting.py [hunk 1]
```diff
@@ -117,3 +117,23 @@
...
+    "targeted": [
+        "MEAN_INSERT_SIZE",
+        "PERCENT_DUPLICATION",
+        "MEAN_TARGET_COVERAGE",
+        "MEDIAN_TARGET_COVERAGE",
+        "PCT_TARGET_BASES_50X",
+        "PCT_TARGET_BASES_100X",
+        "PCT_TARGET_BASES_250X",
+        "PCT_TARGET_BASES_500X",
+        "PCT_TARGET_BASES_1000X",
```
Comment (ashwini06) [inline BALSAMIC/constants/quality_check_reporting.py:131]: Check if you are reporting `PCT_TArget` metrics?

File: tests/test_data/qc_files/analysis/qc/multiqc_data/multiqc_data.json [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+            },
+            "InsertSizeMetrics": {
+                "concatenated_tumor_XXXXXX_R": "tests/test_data/qc_files/analysis/bam/concatenated_tumor_XXXXXX_R.sorted.insertsizemetric"
+            },
+            "DuplicationMetrics": {
+                "concatenated_tumor_XXXXXX_R": "tests/test_data/qc_files/analysis/bam/concatenated_tumor_XXXXXX_R.sorted.mrkdup.txt"
+            }
+        }
+    },
+    "report_saved_raw_data": {
```
Comment (ashwini06) [inline tests/test_data/qc_files/analysis/qc/multiqc_data/multiqc_data.json:16]: IF possible, minimize the long JSON by considering only relevant keys in it.

**Review by ashwini06 — COMMENTED** [review]
Looks good..some of the comments from #800 can go in here as well w.r.t which metrics to consider. You can edit both PRs on the same ground and we can merge it then.

File: BALSAMIC/commands/report/deliver.py [hunk 3]
```diff
@@ -117,7 +125,7 @@ def deliver(
     case_name = sample_config_dict["analysis"]["case_id"]
     result_dir = get_result_dir(sample_config_dict)
     dst_directory = os.path.join(result_dir, "delivery_report")
-    LOG.info("Creatiing delivery_report directory")
+    LOG.info("Creating delivery_report directory")
     os.makedirs(dst_directory, exist_ok=True)
```
[the remark this answers is not in the document]
Comment (Mropat) [reply BALSAMIC/commands/report/deliver.py:129]: https://docs.python.org/3/library/pathlib.html#pathlib.Path.mkdir

**Comment by ivadym** [comment]
Updated! Thanks for your comments @Mropat!

File: BALSAMIC/utils/qc_metrics.py [hunk 4]
```diff
@@ -56,3 +59,56 @@ def get_qc_metrics_json(analysis_path, sequencing_type):
     )
 
     return qc_check_model.get_json
+
+
+def get_multiqc_data_source(data, sample, source_name):
+    """Extracts the metrics data source associated with sample and source names"""
+    source = source_name[:-1].split("_")
```
Comment (hassanfa) [inline BALSAMIC/utils/qc_metrics.py:66]: Is it possible to add some comment what is exactly being extracted by this split?
e.g.`#multiqc_picard_hsmetrics --> XXX`

**Review by hassanfa — APPROVED** [review]
Amazing code as always

File: BALSAMIC/utils/qc_metrics.py (hunk 4 above)
Comment (ivadym) [reply BALSAMIC/utils/qc_metrics.py:66]: Updated! Hopefully it will be clearer now. Thanks for your comments!

**Pull request merged**